lib/r10k/api/modules.rb

```diff
@@ -123,12 +125,19 @@ def write_module(module_name, env_map, path, opts={})
             git.clean(path, git_dir: git_dir, force: true)
           end
         when :forge
-          tarball_cachedir = cachedir_for_forge_module(mod[:source], opts[:cachedir])
+          release_slug = release_slug_from_module_slug_version(mod[:source], mod[:resolved_version])
+          cachedir = opts[:cachedir] 
+          unpackdir = File.join(cachedir, 'unpack')
+          if !File.directory?(unpackdir)
+            FileUtils.mkdir_p(unpackdir)
+          end
 
-          # TODO: cache tarball
+          forge_opts = opts[:forge] || {}
+          release_tarball = get_cached_forge_release(release_slug, cachedir, false, forge_opts)
```

Comment:
Do we need to set the force option from opts? I guess it depends what we are "forcing"